notedetector.py

```diff
@@ -1,92 +1,54 @@
-        self.lastX = self.config["DEFAULT_BBOX"]
-        self.lastY = self.config["DEFAULT_BBOX"]
-        self.lastCounter = 0
-        self.sameCounterCounter = 0
-        self.sameCounterBBox = 0
-        NetworkTables.initialize(server=self.config["NETWORKTABLES_IP"])
-        self.noteSub = NetworkTables.getTable('noteDetector')
-    
-    # get the camera pixel coordinates of the bounding box
-    def getXMin(self):
-        return self.noteSub.getNumber('xmin', self.config["DEFAULT_BBOX"])
-
-    def getXMax(self):
-        return self.noteSub.getNumber('xmax', self.config["DEFAULT_BBOX"])
-
-    def getYMin(self):
-        return self.noteSub.getNumber('ymin', self.config["DEFAULT_BBOX"])
-
-    def getYMax(self):
-        return self.noteSub.getNumber('ymax', self.config["DEFAULT_BBOX"])
-
-    def getCounter(self):
-        # the 'counter' value on the coral should increase by 1 each cycle
-        return self.noteSub.getNumber('counter', 0)
+        self.limelight = NetworkTables.getTable('limelight-note')
+        self.lastHeartbeat = 0
+        self.sameCounter = 0
 
     def hasTarget(self):
-        if self.trustCoral():
-            return bool(self.noteSub.getBoolean('hasTarget', False))
+        if self.trustLimelight():
+            return bool(self.limelight.getNumber('tv', 0))
         else:
             return False
 
-    def trustCoral(self):
-        #return False
-        return self.isAlive()
+    def getTargetAngleX(self):
+        return self.limelight.getNumber('tx', self.config['DEFAULT_BBOX'])
 
-    def isAlive(self):
-        return bool(self.noteSub.getBoolean('isAlive', False))
+    def getTargetAngleZ(self):
+        return self.limelight.getNumber('ty', self.config['DEFAULT_BBOX']) + self.config['CAMERA_ANGLE_ABOVE_HORIZONTAL']
 
-    def testCounter(self):
-        if self.getCounter() > self.lastCounter:
-            self.sameCounterCounter = 0
-        else:
-            self.sameCounterCounter += 1
-
-        self.lastCounter = self.getCounter()
-
-        if self.sameCounterCounter > 3:
-            return False
-        else:
-            return True
-        
-    def testBBox(self):
-        if self.getXMax == self.lastX and self.getYMax == self.lastY and self.lastX != self.config["DEFAULT_BBOX"]:
-            self.sameCounterBBox += 1
-        else:
-            self.sameCounterBBox = 0
+    def getHeartbeat(self):
+        return self.limelight.getNumber('hb', 0)
 
-        self.lastX = self.getXMax
-        self.lastY = self.getYMax
+    def getTargetErrorX(self):
+        return self.getTargetErrorY() * math.tan(math.radians(self.getTargetAngleX())) + self.config['CAMERA_OFFSET_X']
 
-        if self.sameCounterBBox > 10:    
-            return False
-        else:
-            return True
+    def getTargetErrorY(self):
+        return ((self.config['CAMERA_HEIGHT'] - self.config['NOTE_HEIGHT'])/math.tan(math.radians(self.getTargetAngleZ())))*(-1) + self.config['CAMERA_OFFSET_Y']
```

Comment:
I'd suggest breaking this expression into a set of named variables for clarity and then manipulate the variables.